src/main/kotlin/com/terraformation/backend/accelerator/db/CohortStore.kt

```diff
@@ -0,0 +1,122 @@
+package com.terraformation.backend.accelerator.db
+
+import com.terraformation.backend.accelerator.model.CohortModel
+import com.terraformation.backend.accelerator.model.ExistingCohortModel
+import com.terraformation.backend.accelerator.model.NewCohortModel
+import com.terraformation.backend.accelerator.model.toModel
+import com.terraformation.backend.auth.currentUser
+import com.terraformation.backend.customer.model.requirePermissions
+import com.terraformation.backend.db.asNonNullable
+import com.terraformation.backend.db.default_schema.CohortId
+import com.terraformation.backend.db.default_schema.tables.daos.CohortsDao
+import com.terraformation.backend.db.default_schema.tables.pojos.CohortsRow
+import com.terraformation.backend.db.default_schema.tables.references.COHORTS
+import com.terraformation.backend.db.default_schema.tables.references.PARTICIPANTS
+import jakarta.inject.Named
+import java.time.InstantSource
+import org.jooq.Condition
+import org.jooq.DSLContext
+import org.jooq.impl.DSL
+import org.springframework.dao.DataIntegrityViolationException
+
+@Named
+class CohortStore(
+    private val clock: InstantSource,
+    private val dslContext: DSLContext,
+    private val cohortsDao: CohortsDao,
+) {
+  fun fetchOneById(cohortId: CohortId): ExistingCohortModel {
+    return fetch(COHORTS.ID.eq(cohortId)).firstOrNull() ?: throw CohortNotFoundException(cohortId)
+  }
+
+  fun findAll(): List<ExistingCohortModel> {
+    return fetch(null)
+  }
+
+  fun create(model: NewCohortModel): ExistingCohortModel {
+    requirePermissions { createCohort() }
+
+    val now = clock.instant()
+    val userId = currentUser().userId
+
+    val row =
+        CohortsRow(
+            createdBy = userId,
+            createdTime = now,
+            modifiedBy = userId,
+            modifiedTime = now,
+            name = model.name,
+            phaseId = model.phase)
+
+    cohortsDao.insert(row)
+
+    return row.toModel()
+  }
+
+  fun delete(cohortId: CohortId) {
+    requirePermissions { deleteCohort(cohortId) }
+
+    try {
+      dslContext.transaction { _ ->
```

Comment:
This doesn't need to be in a transaction since it's only one SQL statement.

Reply:
I think the transaction is needed, without it, I get `org.jooq.exception.DataAccessException` thrown in the catch block where we are checking for participants (`val hasParticipants = dslContext.fetchExists(PARTICIPANTS, PARTICIPANTS.COHORT_ID.eq(cohortId))`). Or do I need to reset the context somehow so it knows that the previous error is OK to ignore? 

```
Caused by: org.jooq.exception.DataAccessException: SQL [select exists (select 1 as "one" from "participants" where "participants"."cohort_id" = ?)]; ERROR: current transaction is aborted, commands ignored until end of transaction block
```

Reply:
You're right; I didn't think about the error case leaving the (implicit) open transaction in a dirty state.